### PR TITLE
feat(getTemplateOptions): add lifecycle methods

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
The original use case for this was to set up [log4brains](https://github.com/thomvaill/log4brains) after installing, but before committing.

### CODE REVIEW

File: src/generate-from-template.js [hunk 1]
```diff
@@ -26,6 +26,18 @@ const getPackageName = require('./utils/get-package-name');
 const getPackageVersion = require('./utils/get-package-version');
 const { getStoredValues, setStoreValues } = require('./utils/storage');
 
+const noop = () => 0;
```
Comment (code-forger) [inline src/generate-from-template.js:29]: The docs state that these methods should return true to indicate that a step can be skipped. Can this noon return `false` to keep a consistent type?

File: README.md [hunk 2]
```diff
@@ -130,6 +130,19 @@ These values allow you to configure the generator.
 * `initialCommitOptions`: Array of additional git options to be passed to the initial commit.
 * `storeResponses`: If this is set to `true`, create-using-template will store any responses for your template for future template generations.
 
+##### `lifecycle` object<function>, optional
+
+These functions will run at specific stages of the generation lifecycle<!-- .slide: data-fullscreen -->
```
Comment (Matthew-Mallimo) [inline README.md:135]: What does `<!-- .slide: data-fullscreen -->` do?

Comment (10xLaCroixDrinker) [reply README.md:135]: I have no idea where that comment came from. I didn't type that.

File: src/generate-from-template.js (hunk 1 above)
Comment (10xLaCroixDrinker) [reply src/generate-from-template.js]: done

File: README.md (hunk 2 above)
Comment (10xLaCroixDrinker) [reply README.md]: removed

**Review by Matthew-Mallimo — DISMISSED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**